docs/content/docs/connectors/datastream/kafka.md

```diff
@@ -613,134 +613,89 @@ Otherwise, the watermarks of the whole application cannot advance and all time-b
-2. A SerializationSchema / KafkaSerializationSchema for serializing data into Kafka
-3. Properties for the Kafka client. The following properties are required:
-    * "bootstrap.servers" (comma separated list of Kafka brokers)
-4. A fault-tolerance semantic
+Kafka sink provides a builder class to construct an instance of a KafkaSink. The code snippet below
+shows how to write String records to a Kafka topic with a delivery guarantee of at least once.
 
-{{< tabs "8dbb32b3-47e8-468e-91a7-43cec0c658ac" >}}
-{{< tab "Java" >}}
 ```java
 DataStream<String> stream = ...
+        
+KafkaSink<String> sink = KafkaSink.<String>builder()
+        .setBootstrapServers(brokers)
+        .setRecordSerializer(KafkaRecordSerializationSchema.builder()
+            .setTopic("topic-name")
+            .setKafkaProducerConfig(producerProperties)
+            .setValueSerializationSchema(new SimpleStringSchema())
+            .setDeliveryGuarantee(DeliveryGuarantee.AT_LEAST_ONCE)
+            .build()
+        )
+        .build();
+        
+stream.sinkTo(sink);
+```
 
-Properties properties = new Properties();
-properties.setProperty("bootstrap.servers", "localhost:9092");
+The following properties are **required** to build a KafkaSink:
+
+- Bootstrap servers, ```setBootstrapServers(String)```
+- Record serializer, ``setRecordSerializer(KafkaRecordSerializationSchema)``
+- Kafka producer properties, ```setKafkaProducerConfig(Properties)```
+- If you configure the delivery guarantee with ```DeliveryGuarantee.EXACTLY_ONCE``` you also have
+  use ```setTransactionalIdPrefix(String)```
 
-FlinkKafkaProducer<String> myProducer = new FlinkKafkaProducer<>(
-        "my-topic",                  // target topic
-        new SimpleStringSchema(),    // serialization schema
-        properties,                  // producer config
-        FlinkKafkaProducer.Semantic.EXACTLY_ONCE); // fault-tolerance
+### Serializer
 
-stream.addSink(myProducer);
+You always need construct a ```KafkaRecordSerializationSchema``` to transform incoming elements from
+the data stream to Kafka producer records.
+We offer a schema builder to provide some common building blocks i.e. key/value serialization, topic
+selection, partitioning.
+
+```java
+KafkaRecordSerializationSchema.builder()
+    .setTopicSelector((element) -> {<your-topic-selection-logic>})
+    .setValueSerializationSchema(new SimpleStringSchema())
+    .setKeySerializationSchema(new SimpleStringSchema())
+    .setPartitioner(new FlinkFixedPartitioner())
+    .build();
 ```
-{{< /tab >}}
-{{< tab "Scala" >}}
-```scala
-val stream: DataStream[String] = ...
 
-val properties = new Properties
-properties.setProperty("bootstrap.servers", "localhost:9092")
+It is **required** to always set a value serialization method and a topic retrieval method.
+Moreover, it is also possible to use Kafka serializers instead of Flink serializer by using 
+```setKafkaKeySerializer(Serializer)``` or ```setKafkaValueSerializer(Serializer)```.
```

Comment:
```suggestion
It is **required** to always set a value serialization method and a topic (selection method).
Moreover, it is also possible to use Kafka serializers instead of Flink serializer by using 
```setKafkaKeySerializer(Serializer)``` or ```setKafkaValueSerializer(Serializer)```.
```